File: CS/Reporting_report-storage-for-the-end-user-report-designer-e2704/StorageEditorForm.cs
using System;
using System.Windows.Forms;
// ...

namespace ReportStorageSample {
    public partial class StorageEditorForm : Form {
        public StorageEditorForm() {
            InitializeComponent();
        }

        private void listBox1_SelectedIndexChanged(object sender, EventArgs e) {
            textBox1.Text = listBox1.SelectedItem.ToString();
        }

        private void StorageEditorForm_Load(object sender, EventArgs e) {
            if (listBox1.Items.Count > 0 && string.IsNullOrEmpty(textBox1.Text))
                listBox1.SelectedIndex = 0;
        }

        private void textBox1_TextChanged(object sender, EventArgs e) {
            buttonOK.Enabled = !string.IsNullOrEmpty(textBox1.Text);
        }

        private void buttonOK_Click(object sender, EventArgs e) {

        }
    }
}

File: VB/Reporting_report-storage-for-the-end-user-report-designer-e2704/StorageEditorForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
' ...

Namespace ReportStorageSample
	Partial Public Class StorageEditorForm
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub listBox1_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles listBox1.SelectedIndexChanged
			textBox1.Text = listBox1.SelectedItem.ToString()
		End Sub

		Private Sub StorageEditorForm_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			If listBox1.Items.Count > 0 AndAlso String.IsNullOrEmpty(textBox1.Text) Then
				listBox1.SelectedIndex = 0
			End If
		End Sub

		Private Sub textBox1_TextChanged(ByVal sender As Object, ByVal e As EventArgs) Handles textBox1.TextChanged
			buttonOK.Enabled = Not String.IsNullOrEmpty(textBox1.Text)
		End Sub

		Private Sub buttonOK_Click(ByVal sender As Object, ByVal e As EventArgs) Handles buttonOK.Click

		End Sub
	End Class
End Namespace
